### PR TITLE
Add caching

### PR DESCRIPTION
**Pull request by cammellos**
fixes https://github.com/status-im/status-mobile/issues/20975

This commit integrates caching of the token balances.

To test:

1) Login when connection is working, make sure you have all the balances.
2) Logout, go offline
3) Login again

You should see the amount of tokens you have (unless it's less that 0.000001, that will be displayed as 0, but we could show it). 
Price in USD will not be shown as price is not cached yet.

### CODE REVIEW

**Comment by churik** [comment]
@cammellos Based on our pair testing caching / routes building and in general interaction with wallet feels more robust than in current release + a way less "Provider is down" errors (your PR doesn't include commit on hiding it)

Thank you for digging into it, a always EXCELLENT WORK! 🏅

File: src/status_im/subs/wallet/wallet.cljs [hunk 1]
```diff
@@ -340,10 +340,14 @@
  :<- [:profile/currency]
  :<- [:wallet/selected-networks->chain-ids]
  (fn [[accounts currency chain-ids]]
+   (println "CHAINS" currency chain-ids)
```
Comment (shivekkhurana) [inline src/status_im/subs/wallet/wallet.cljs]: Rogue println

File: src/status_im/contexts/wallet/events.cljs [hunk 2]
```diff
@@ -190,7 +190,9 @@
     :fx [[:json-rpc/call
           [{:method     "wallet_getWalletToken"
             :params     [[address]]
-            :on-success [:wallet/store-wallet-token address]
+            :on-success #(do
+                           (println "RESPONSE HERE" %)
```
Comment (shivekkhurana) [inline src/status_im/contexts/wallet/events.cljs]: Rogue println

**Review by shivekkhurana — CHANGES_REQUESTED** [review]

File: src/status_im/config.cljs [hunk 3]
```diff
@@ -48,7 +48,7 @@
 
 (def opensea-api-key OPENSEA_API_KEY)
 (def status-proxy-enabled? true)
-(def status-proxy-stage-name (get-config :STATUS_PROXY_STAGE_NAME "test"))
+(def status-proxy-stage-name (get-config :STATUS_PROXY_STAGE_NAME "prod"))
```
Comment (J-Son89) [inline src/status_im/config.cljs]: intentional??

Comment (cammellos) [reply src/status_im/config.cljs]: yes, but not meant to be merged, but qa will want to test against prod (same as release)

**Comment by ilmotta** [comment]
@churik @cammellos is this the behavior we expect? I followed the instructions in the PR description. After all 3 steps, after login the balances are not displayed, only the loading skeletons.

https://github.com/user-attachments/assets/4472e95b-8daf-473b-a121-9aafbf7f7a1d

**Comment by cammellos** [comment]
mmh, it should eventually resolve the rpc call, but if you are running the last build it's experimental @ilmotta , try the one next to last (number 3)

**Comment by ilmotta** [comment]
False negative, worked well @cammellos. Sorry about that

**Review by ilmotta — APPROVED** [review]

**Review by shivekkhurana — APPROVED** [review]

**Comment by churik** [comment]
Looks good for me, ready to cherry-pick and merge

**Pull request closed without merging**